### PR TITLE
Checkpointing, NetCDF, and golden master tests

### PR DESCRIPTION
**Pull request by ali-ramadhan**
Call them system tests, model verification tests, or end-to-end tests, that's what this PR is about.

Currently a work in progress as I hit a roadblock with checkpointing so **the tests will fail**. See #141. BUT, for testing we can just set `model.forcings = nothing` and manually reconstruct the forcings as we know them.

Tests implemented so far:
1. Checkpointing integration test: Run two coarse rising thermal bubble simulations and make sure that when restarting from a checkpoint, the restarted simulation matches the non-restarted simulation numerically.
2. NetCDF output integration test: Run a coarse thermal bubble simulation and save the output to NetCDF at the 10th time step. Then read back the output and test that it matches the model's state.
3. Thermal bubble golden master test: Run the coarse thermal bubble simulation for 10 time steps and check that the model output matches a golden master output.
4. Deep convection golden master test

@jm-c @christophernhill: do post any ideas for tests that should be implemented in this PR.

### CODE REVIEW

File: src/output_writers.jl [hunk 1]
```diff
@@ -44,25 +48,34 @@ filename(fw::Checkpointer, name, iteration) = filename(fw, "model_checkpoint", i
...
     println("[Checkpointer] Serializing model to disk: $filepath")
     f = JLD.jldopen(filepath, "w", compress=true)
     JLD.@write f model
     close(f)
 
-    # Reconstruct PoissonSolver struct with FFT plans ?
+    println("[Checkpointer] Reconstructing FFT plans...")
     metadata, grid, stepper_tmp = model.metadata, model.grid, model.stepper_tmp
-    if metadata.arch == :cpu
+    if metadata.arch == :CPU
```
Comment (glwagner) [inline src/output_writers.jl]: Why do we save FFT plans?

It makes sense that a user should plan new FFTs if they are going to run a model, even if they are restarting a checkpoint on the same computer.

@christophernhill what do you think?

Comment (ali-ramadhan) [reply src/output_writers.jl]: We don't. We set `model.poisson_solver = nothing` before checkpointing then reconstruct the FFT plans afterwards.

So you're saying the user should be manually planning the FFTs themselves?

**Comment by ali-ramadhan** [comment]
**Note to self**: Seems like the golden master tests fail on Mac (Travis CI) so it seems that JLD might not be portable (although I generated the JLD checkpoint files on Windows and the Linux tests pass).

So for golden master testing I should rely on NetCDF output which is guaranteed to be portable.

**Comment by ali-ramadhan** [comment]
@jm-c and I discussed this PR yesterday and seemed good to merge. Looks like the tests work now so I'm merging.

**Pull request merged**